runtime/compiler/aarch64/codegen/CallSnippet.cpp

```diff
@@ -0,0 +1,590 @@
+   TR::SymbolReference *methodSymRef = getNode()->getSymbolReference();
+   TR::SymbolReference *glueRef = cg()->symRefTab()->findOrCreateRuntimeHelper(TR_ARM64interfaceCallHelper, false, false, false);
+
+   getSnippetLabel()->setCodeLocation(cursor);
+
+   // bl glueRef
+   *(int32_t *)cursor = encodeHelperBranchAndLink(glueRef, cursor, getNode(), cg());
+   cursor += 4;
+
+   // Store the code cache RA
+   *(intptrj_t *)cursor = (intptrj_t)getReturnLabel()->getCodeLocation();
+   cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(
+                               cursor,
+                               NULL,
+                               TR_AbsoluteMethodAddress, cg()),
+                               __FILE__, __LINE__, getNode());
+   cursor += 8;
+
+   // CP
+   *(intptrj_t *)cursor = (intptrj_t)methodSymRef->getOwningMethod(cg()->comp())->constantPool();
+   cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(
+                               cursor,
+                               *(uint8_t **)cursor,
+                               getNode() ? (uint8_t *)getNode()->getInlinedSiteIndex() : (uint8_t *)-1,
+                               TR_Thunks, cg()),
+                               __FILE__, __LINE__, getNode());
+   cursor += 8;
+
+   // CP index
+   *(int32_t *)cursor = methodSymRef->getCPIndexForVM();
+   cursor += 4;
+
+   // Add 2 more slots for resolved values
+   *(int32_t *)cursor = 0;
+   cursor += 4;
+   *(int32_t *)cursor = 0;
+   cursor += 4;
+
+   return cursor;
+   }
+
+uint32_t TR::ARM64InterfaceCallSnippet::getLength(int32_t estimatedSnippetStart)
+   {
+   return 32;
+   }
+
+uint8_t *TR::ARM64CallSnippet::generateVIThunk(TR::Node *callNode, int32_t argSize, TR::CodeGenerator *cg)
+   {
+   int32_t codeSize = 4 * (instructionCountForArguments(callNode, cg) + 5) + 8; // 5 instructions for branch, Additional 8 bytes to hold size of thunk
+   uint8_t *thunk, *buffer, *returnValue;
+   intptr_t dispatcher;
+
+   if (cg->comp()->compileRelocatableCode())
+      thunk = (uint8_t *)TR::comp()->trMemory()->allocateMemory(codeSize, heapAlloc);
+   else
+      thunk = (uint8_t *)cg->allocateCodeMemory(codeSize, true, false);
+   buffer = returnValue = thunk + 8;
+   TR_RuntimeHelper helper;
+   TR::DataType dataType = callNode->getDataType();
+
+   switch (dataType)
+      {
+      case TR::NoType:
+         helper = TR_ARM64icallVMprJavaSendVirtual0;
+         break;
+      case TR::Int32:
+         helper = TR_ARM64icallVMprJavaSendVirtual1;
+         break;
+      case TR::Int64:
+      case TR::Address:
+         helper = TR_ARM64icallVMprJavaSendVirtualJ;
+         break;
+      case TR::Float:
+         helper = TR_ARM64icallVMprJavaSendVirtualF;
+         break;
+      case TR::Double:
+         helper = TR_ARM64icallVMprJavaSendVirtualD;
+         break;
+      default:
+         TR_ASSERT(false, "Bad return data type '%s' for a call node.\n",
```

Comment:
A fatal assert is probably warranted here.

Reply:
Changed to `TR_ASSERT_FATAL()`.